src/dead_letters.rs

```diff
@@ -286,9 +287,10 @@ impl DeadLetterQueue for DeltaSinkDeadLetterQueue {
             .map(|dl| {
                 serde_json::to_value(dl)
                     .map_err(|e| DeadLetterQueueError::SerdeJson { source: e })
-                    .and_then(|mut v| {
+                    .and_then(|v| {
                         self.transformer
-                            .transform(&mut v, None as Option<&BorrowedMessage>)?;
+                            // TODO: this can't be right, shouldn't this function takje DeserializedMessage
+                            .transform(&mut v.clone().into(), None as Option<&BorrowedMessage>)?;
```

Comment:
I don't think this does what is intended. The transform function takes a mutable reference to the clone of `v` instead of a mutable reference to `v` itself. The function modifies and consumes the clone, but it only exists in the scope of the input to `transform`, so it should be dropped once the method finishes. That means `Ok(v)` is returning the original instead of the transformed message. Implementing `Into<Value>` for `DeserializedMessage` and changing the code to
```
let mut dm = DeserializedMessage::from(v);
                        self.transformer
                            // TODO: this can't be right, shouldn't this function takje DeserializedMessage
                            .transform(&mut dm, None as Option<&BorrowedMessage>)?;
                        
                        Ok(dm.into())
```
should do the trick. Also the `Into` implementation will allow for the removal of all of instances of `to_owned()` that my last commit added.

Reply:
The transform code might be some bungled merge conflict resolution, I'll be able to look more into this shortly